tests/systemd_testsuite/test_01_basic.pm

```diff
@@ -18,7 +18,36 @@ use utils 'zypper_call';
 use power_action_utils 'power_action';
 
 sub run {
-    select_console 'root-console';
+    #prepare test
+    assert_script_run 'cd /var/opt/systemd-tests';
+    assert_script_run './run-tests.sh TEST-01-BASIC --setup 2>&1 | tee /tmp/testsuite.log', 600;
+    assert_script_run 'ls -l /etc/systemd/system/testsuite.service';
+    #reboot
+    power_action('reboot', textmode => 1);
+    assert_screen('linux-login', 600);
+    type_string "root\n";
+    wait_still_screen 3;
+    type_password;
+    wait_still_screen 3;
+    send_key 'ret';
+    #run test
+    assert_script_run 'cd /var/opt/systemd-tests';
+    assert_script_run './run-tests.sh TEST-01-BASIC --run 2>&1 | tee /tmp/testsuite.log', 60;
+    #cleanup
+    assert_screen("systemd-testsuite-test-01-basic");
+    assert_script_run './run-tests.sh TEST-01-BASIC --clean 2>&1 | tee /tmp/testsuite.log', 60;
 }
 
+sub test_flags {
+    return { always_rollback => 1 };
+}
+
+sub post_fail_hook {
+    my ($self) = shift;
+    $self->SUPER::post_fail_hook;
+    assert_script_run('tar -cjf TEST-01-BASIC-logs.tar.bz2 /var/opt/systemd-tests/logs/');
+    upload_logs('TEST-01-BASIC-logs.tar.bz2');
+}
```

Comment:
The only thing that changes in each test, is the filename. We should extract the post_fail_hook to a base class.

Reply:
Could you give me some example for creating a base class, so that I can adapt the code?

Reply:
I think this is a good example: https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/consoletest.pm

But, instead of **post_run_hook** it would be **pre_run_hook**: http://open.qa/docs/#_how_to_write_tests

Reply:
The hierarchy would be:

```
lib/consoletest.pm
|
'---- lib/systemd_testsuite_test.pm
       |
       '---- systemd_testsuite/test_01_basic.pm
```